plugin/lspconfig.lua

```diff
@@ -3,129 +3,129 @@ if vim.g.lspconfig ~= nil then
-  for _, config in ipairs(matching_configs) do
-    config.launch()
-  end
 end, {
   desc = 'Manually launches a language server',
   nargs = '?',
-  complete = lsp_complete_configured_servers,
+  complete = complete_config,
 })
 
-api.nvim_create_user_command('LspRestart', function(info)
+vim.api.nvim_create_user_command('LspRestart', function(info)
+  local clients = vim
+    .iter(info.fargs)
+    :map(function(name)
+      local client = vim.lsp.get_clients({ name = name })[1]
+      if client == nil then
+        vim.notify(("Invalid server name '%s'"):format(name))
+      end
+      return client
+    end)
+    :totable()
+
   local detach_clients = {}
-  for _, client in ipairs(get_clients_from_cmd_args(info.args)) do
-    -- Can remove diagnostic disabling when changing to client:stop() in nvim 0.11+
-    --- @diagnostic disable: missing-parameter
-    client.stop()
-    if vim.tbl_count(client.attached_buffers) > 0 then
-      detach_clients[client.name] = { client, lsp.get_buffers_by_client_id(client.id) }
-    end
+  for _, client in ipairs(clients) do
+    detach_clients[vim.lsp.config[client.name]] = vim.lsp.get_buffers_by_client_id(client.id)
+    client:stop()
   end
+
   local timer = assert(vim.uv.new_timer())
   timer:start(
     500,
-    100,
+    0,
     vim.schedule_wrap(function()
-      for client_name, tuple in pairs(detach_clients) do
-        if require('lspconfig.configs')[client_name] then
-          local client, attached_buffers = unpack(tuple)
-          if client.is_stopped() then
-            for _, buf in pairs(attached_buffers) do
-              require('lspconfig.configs')[client_name].launch(buf)
-            end
-            detach_clients[client_name] = nil
+      for config, buffers in pairs(detach_clients) do
+        for _, bufnr in ipairs(buffers) do
+          if type(config.root_dir) == 'function' then
+            ---@param root_dir string
+            config.root_dir(bufnr, function(root_dir)
+              config.root_dir = root_dir
+              vim.schedule(function()
+                vim.lsp.start(config, {
+                  bufnr = bufnr,
+                  reuse_client = config.reuse_client,
+                  _root_markers = config.root_markers,
+                })
+              end)
+            end)
+          else
+            vim.lsp.start(config, {
+              bufnr = bufnr,
+              reuse_client = config.reuse_client,
+              _root_markers = config.root_markers,
+            })
           end
         end
       end
-
-      if next(detach_clients) == nil and not timer:is_closing() then
-        timer:close()
-      end
     end)
   )
 end, {
   desc = 'Manually restart the given language client(s)',
-  nargs = '?',
-  complete = lsp_get_active_clients,
+  nargs = '*',
+  complete = complete_client,
 })
 
-api.nvim_create_user_command('LspStop', function(info)
+vim.api.nvim_create_user_command('LspStop', function(info)
```

Comment:
Nvim 0.11+, `LspStop` can just do this?:

    vim.lsp.enable(..., false)

Reply:
Yes, I'll add it. Should I re-enable the config in `LspStart`?